lib/action_subscriber/middleware/env.rb

```diff
@@ -95,7 +107,7 @@ def to_hash
     private
 
       def has_used_delivery_tag?
-        @has_been_acked || @has_been_nacked || @has_been_rejected
+        @has_been_acked || @has_been_nacked || @has_been_rejected || @disable_safe_actions
```

Comment:
If we're going with this config option, shouldn't that check be first here?

Reply:
Doesn’t matter since any one of them can be true. We use this method to signal an action has already been performed. The safe methods do nothing if any of these instance variables are set to true.